app/models/host/managed.rb

```diff
@@ -822,14 +854,26 @@ def available_template_kinds(provisioning = nil)
             end
 
     kinds.map do |kind|
-      ProvisioningTemplate.find_template({ :kind               => kind.name,
```

Comment:
It's related. I want aspects be able to add filter options based on the information in the aspect.
I have added call to `#template_filter_from_aspects` which checks each of the aspects if it responds to `#template_filter_options`, and if it does - it can add another filtering.
You can see it implemented in the reference PR for the puppet environment: 
Base filter (without environment): https://github.com/theforeman/foreman/pull/2555/files#diff-2d53894341fd1cf84a823e3bd0855495R716
The addition (filter by puppet environment): https://github.com/theforeman/foreman/pull/2555/files#diff-21d16583cdd439b78dfb8ee84c3fc406R25